test/controllers/comment_controller_test.exs

```diff
@@ -64,6 +64,28 @@ defmodule CodeCorps.CommentControllerTest do
     test "does not create resource and renders 403 when not authorized", %{conn: conn} do
       assert conn |> request_create(@valid_attrs) |> json_response(403)
     end
+
+    @tag :authenticated
+    test "tracks comment created", %{conn: conn, current_user: current_user} do
+      task = insert(:task)
+      attrs = @valid_attrs |> Map.merge(%{task: task, user: current_user})
+      comment_id = conn
+                    |> request_create(attrs)
+                    |> json_response(201)
+                    |> get_in(["data", "id"])
+                    |> String.to_integer
+
+      id = current_user.id
+      properties = %{
+        comment_id: comment_id,
+        task: task.title,
+        task_id: task.id,
+        task_type: task.task_type,
+        project_id: task.project_id
+      }
+
+      assert_received {:track, ^id, "Created Comment", ^properties}
```

Comment:
I would have thought so too. But I learned something new when I tried this earlier. 

First, as you suggested, with `^current_user.id`:

```bash
$ docker-compose run --rm test mix test test/controllers/comment_controller_test.exs:69
Starting codecorpsapi_elasticsearch_1
Including tags: [line: "69"]
Excluding tags: [:test]

** (CompileError) test/controllers/comment_controller_test.exs:87: invalid argument for unary operator ^, expected an existing variable, got: ^current_user.id()
    (stdlib) lists.erl:1354: :lists.mapfoldl/3
    (stdlib) lists.erl:1355: :lists.mapfoldl/3
    (ex_unit) expanding macro: ExUnit.Assertions.assert_received/1
    test/controllers/comment_controller_test.exs:87: CodeCorps.CommentControllerTest."test create tracks comment created"/1
    (elixir) lib/code.ex:363: Code.require_file/2
    (elixir) lib/kernel/parallel_require.ex:57: anonymous fn/2 in Kernel.ParallelRequire.spawn_requires/5

01:57:20.252 [error] GenServer #PID<0.228.0> terminating
** (CompileError) test/controllers/comment_controller_test.exs:87: invalid argument for unary operator ^, expected an existing variable, got: ^current_user.id()
    (stdlib) lists.erl:1354: :lists.mapfoldl/3
    (stdlib) lists.erl:1355: :lists.mapfoldl/3
    (ex_unit) expanding macro: ExUnit.Assertions.assert_received/1
    test/controllers/comment_controller_test.exs:87: CodeCorps.CommentControllerTest."test create tracks comment created"/1
    (elixir) src/elixir.erl:223: :elixir.erl_eval/3
    (elixir) src/elixir.erl:211: :elixir.eval_forms/4
    (elixir) src/elixir_compiler.erl:66: :elixir_compiler.eval_compilation/3
    (elixir) src/elixir_lexical.erl:17: :elixir_lexical.run/3
    (elixir) src/elixir_compiler.erl:30: :elixir_compiler.quoted/3
    (elixir) lib/code.ex:363: Code.require_file/2
    (elixir) lib/kernel/parallel_require.ex:57: anonymous fn/2 in Kernel.ParallelRequire.spawn_requires/5
```

We can only use the pin operator with an existing variable, not as the result of a function call.

Just for giggles, let's try it without the pin operator, as `current_user.id`:

```bash
ross@plato [08:57:21][~/dev/code-corps/code-corps-api][test_tracking_in_controllers *]
$ docker-compose run --rm test mix test test/controllers/comment_controller_test.exs:69
Starting codecorpsapi_elasticsearch_1
Including tags: [line: "69"]
Excluding tags: [:test]

** (CompileError) test/controllers/comment_controller_test.exs:87: cannot invoke remote function current_user.id/0 inside match
    (stdlib) lists.erl:1354: :lists.mapfoldl/3
    (stdlib) lists.erl:1355: :lists.mapfoldl/3
    (stdlib) lists.erl:1354: :lists.mapfoldl/3
    (stdlib) lists.erl:1354: :lists.mapfoldl/3

01:58:00.671 [error] GenServer #PID<0.228.0> terminating
** (CompileError) test/controllers/comment_controller_test.exs:87: cannot invoke remote function current_user.id/0 inside match
    (elixir) src/elixir_translator.erl:232: :elixir_translator.translate/2
    (stdlib) lists.erl:1354: :lists.mapfoldl/3
    (stdlib) lists.erl:1355: :lists.mapfoldl/3
    (elixir) src/elixir_translator.erl:23: :elixir_translator.translate/2
    (elixir) src/elixir_translator.erl:17: :elixir_translator.translate/2
    (stdlib) lists.erl:1354: :lists.mapfoldl/3
    (elixir) src/elixir_clauses.erl:26: :elixir_clauses.match/3
    (elixir) src/elixir_clauses.erl:35: :elixir_clauses.clause/6
    (elixir) src/elixir_clauses.erl:83: anonymous fn/3 in :elixir_clauses.do_clauses/3
    (stdlib) lists.erl:1354: :lists.mapfoldl/3
    (elixir) src/elixir_clauses.erl:88: :elixir_clauses.do_clauses/3
    (elixir) src/elixir_clauses.erl:73: :elixir_clauses.clauses/3
    (elixir) src/elixir_translator.erl:131: :elixir_translator.translate/2
    (elixir) src/elixir_translator.erl:16: :elixir_translator.translate/2
    (elixir) src/elixir_translator.erl:374: :elixir_translator.translate_block/3
    (elixir) src/elixir_translator.erl:38: :elixir_translator.translate/2
    (elixir) src/elixir_clauses.erl:36: :elixir_clauses.clause/6
    (elixir) src/elixir_def.erl:170: :elixir_def.translate_clause/7
    (elixir) src/elixir_def.erl:159: :elixir_def.translate_definition/8
    (elixir) src/elixir.erl:223: :elixir.erl_eval/3
```

Another error; the args we pass to `assert_received` end up on the [left side of a pattern match](https://github.com/elixir-lang/elixir/blob/e44ec1ecee15093e0f5c556c4237b8e855603c70/lib/ex_unit/lib/ex_unit/assertions.ex#L389) and Elixir won't allow use to invoke a function there.

So we need to extract a var unless you can think of another way.